svcb_test.go

```diff
@@ -0,0 +1,100 @@
+package dns
+
+import (
+	"strings"
+	"testing"
+)
+
+// This tests everything valid about SVCB but parsing.
+// Parsing tests belong to parse_test.go.
+func TestSVCB(t *testing.T) {
+	svcbs := map[string]string{
+		`mandatory=key65000,alpn`:        `key65000,alpn`,
+		`alpn=h2,h2c`:                    `h2,h2c`,
+		`port="499"`:                     `499`,
+		`ipv4hint=3.4.3.2,1.1.1.1`:       `3.4.3.2,1.1.1.1`,
+		`no-default-alpn=`:               ``,
+		`ipv6hint=1::4:4:4:4,1::3:3:3:3`: `1::4:4:4:4,1::3:3:3:3`,
+		`echconfig=Mw==`:                 `Mw==`,
+		`key65000=4\ 3`:                  `4\ 3`,
+		`key65001="\" "`:                 `\"\ `,
+		`key65002=""`:                    ``,
+		`key65003==\"\"`:                 `=\"\"`,
+		`key65004=\254\032\030\000`:      `\254\ \030\000`,
+	}
+
+	for s, o := range svcbs {
+		key := ""
+		val := ""
+		idx := strings.IndexByte(s, '=')
```

Comment:
Is there a good reason to reimplement all of this instead of just using the parser code (eventually hitting `(*SVCB) parse`)?